corporate/views/session.py

```diff
@@ -18,23 +19,14 @@
 @require_billing_access
 @has_request_variables
 def start_card_update_stripe_session(request: HttpRequest, user: UserProfile) -> HttpResponse:
-    customer = get_customer_by_realm(user.realm)
-    assert customer
-    stripe_session = stripe.checkout.Session.create(
-        cancel_url=f"{user.realm.uri}/billing/",
-        customer=customer.stripe_customer_id,
-        metadata={
-            "type": "card_update",
-            "user_id": user.id,
-        },
-        mode="setup",
-        payment_method_types=["card"],
-        success_url=f"{user.realm.uri}/billing/event_status?stripe_session_id={{CHECKOUT_SESSION_ID}}",
-    )
-    Session.objects.create(
-        stripe_session_id=stripe_session.id,
-        customer=customer,
-        type=Session.CARD_UPDATE_FROM_BILLING_PAGE,
+    billing_session = RealmBillingSession(user)
+    assert billing_session.get_customer() is not None
+    metadata = {
+        "type": "card_update",
+        "user_id": user.id,
```

Comment:
This is from the `user: UserProfile` in the request variables, so I thought this `user.id` could be updated if/when the request variables are also updated.

Also the case for the `user.id` added to the metadata in `start_retry_payment_intent_session`.

Reply:
Yeah, that's reasonable. I suspect longer-term we will want to just have the "blob of extra metadata about the user" to use more common code -- I'm not entirely sure what the current policy is as to where we just do a `user_id` vs more full details on the realm.